pkg/prometheus/promcfg.go

```diff
@@ -2995,14 +3004,8 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 			}
 
 			if config.Type != nil {
-				if *config.Type == "NS" {
-					configs[i] = cg.WithMinimumVersion("2.49.0").AppendMapItem(configs[i], "type", config.Type)
-				} else {
-					configs[i] = append(configs[i], yaml.MapItem{
-						Key:   "type",
-						Value: config.Type,
-					})
-				}
+				minimumSupportedVersion := compatibilityMatrix[*config.Type]
```

Comment:
(nit) to avoid listing all types with "fake" versions
```suggestion
				typeCfg := cfg
				minVersion, found := compatibilityMatrix[*config.Type]
				if found {
					typeCfg = typeCfg.WithMinimumVersion(minVersion)
				}
```

Reply:
Thanks, that's a better approach.